packages/rum/src/resourceUtils.ts

```diff
@@ -48,68 +48,122 @@ export function computeResourceKind(timing: PerformanceResourceTiming) {
+}
+
 export function computePerformanceResourceDetails(
-  entry?: PerformanceResourceTiming
+  entry: PerformanceResourceTiming
 ): PerformanceResourceDetails | undefined {
-  if (!entry || !hasTimingAllowedAttributes(entry) || isCached(entry)) {
-    return undefined
-  }
+  const {
+    startTime,
+    fetchStart,
+    domainLookupStart,
+    domainLookupEnd,
+    connectStart,
+    secureConnectionStart,
+    connectEnd,
+    requestStart,
+    responseStart,
+    responseEnd,
+  } = entry
+  let { redirectStart, redirectEnd } = entry
+
+  // Ensure timings are in the right order.  On top of filtering out potential invalid
+  // PerformanceResourceTiming, it will ignore entries from requests where timings cannot be
+  // collected, for example cross origin requests without a "Timing-Allow-Origin" header allowing
+  // it.
   if (
-    !isValidTiming(entry.connectStart, entry.connectEnd) ||
-    !isValidTiming(entry.domainLookupStart, entry.domainLookupEnd) ||
-    !isValidTiming(entry.responseStart, entry.responseEnd) ||
-    !isValidTiming(entry.requestStart, entry.responseStart) ||
-    !isValidTiming(entry.redirectStart, entry.redirectEnd) ||
-    !isValidTiming(entry.secureConnectionStart, entry.connectEnd)
+    !areInOrder(
+      startTime,
+      fetchStart,
+      domainLookupStart,
+      domainLookupEnd,
+      connectStart,
+      connectEnd,
+      requestStart,
+      responseStart,
+      responseEnd
+    )
   ) {
     return undefined
   }
-  return {
-    connect: isRelevantTiming(entry.connectStart, entry.connectEnd, entry.fetchStart)
-      ? formatTiming(entry.connectStart, entry.connectEnd)
-      : undefined,
-    dns: isRelevantTiming(entry.domainLookupStart, entry.domainLookupEnd, entry.fetchStart)
-      ? formatTiming(entry.domainLookupStart, entry.domainLookupEnd)
-      : undefined,
-    download: formatTiming(entry.responseStart, entry.responseEnd),
-    firstByte: formatTiming(entry.requestStart, entry.responseStart),
-    redirect: isRelevantTiming(entry.redirectStart, entry.redirectEnd, 0)
-      ? formatTiming(entry.redirectStart, entry.redirectEnd)
-      : undefined,
-    ssl:
-      entry.secureConnectionStart !== 0 &&
-      isRelevantTiming(entry.secureConnectionStart, entry.connectEnd, entry.fetchStart)
-        ? formatTiming(entry.secureConnectionStart, entry.connectEnd)
-        : undefined,
+
+  // The only time fetchStart is different than startTime is if a redirection occured.
+  const hasRedirectionOccured = fetchStart !== startTime
+
+  if (hasRedirectionOccured) {
+    // Firefox doesn't provide redirect timings on cross origin requests.  Provide a default for
+    // those.
+    if (redirectStart < startTime) {
+      redirectStart = startTime
+    }
+    if (redirectEnd < startTime) {
+      redirectEnd = fetchStart
+    }
+
+    // Make sure redirect timings are in order
+    if (!areInOrder(startTime, redirectStart, redirectEnd, fetchStart)) {
+      return undefined
+    }
```

Comment:
No, because if no redirection occurred, `redirectStart` and `redirectEnd` are expected to be 0, so the `hasRedirectionOccured` condition is needed before testing those.

Reply:
Yes, what I had in mind was something like:
```js
...
let { redirectStart, redirectEnd } = entry
const hasRedirectionOccured = fetchStart !== startTime
if (hasRedirectionOccured) {
...
}
if (
  !areInOrder(
    startTime,
    redirectStart,
    redirectEnd, 
    fetchStart,
    domainLookupStart,
    ...
) {
  return undefined
}
...
```
In order to check the order once with all the timings. Is there anything preventing it?

Reply:
Yes: if `hasRedirectionOccured == false`, `redirectStart` and `redirectEnd` will be `0` (as expected), so `< startTime`, so we would ignore all timings without redirection

Reply:
ok, I get it